### PR TITLE
feature: rootless deployment

### PR DESCRIPTION
**Pull request by jichenjc**
Fixes: #491

### CODE REVIEW

**Comment by jichenjc** [comment]
need update https://sustainable-computing.io/installation/kepler/ later on, basically we can 

default is `false` which means we use root in CI, then if you need non-root, use following 
```
ROOTLESS=true make
```

and generate manifest is 
```
OPTS="ROOTLESS" make build-manifest
```

then we have 

```
        securityContext:
          capabilities:
            add:
            - NET_ADMIN
            - SYS_ADMIN
            - SYS_RESOURCE
          privileged: true
          runAsGroup: 2000
          runAsUser: 2000
          seLinuxOptions:
            level: s0

```
in `_output/generated-manifest/deployment.yaml`

**Comment by jichenjc** [comment]
@rootfs FYI

**Comment by jichenjc** [comment]
currently we are using 
```
- NET_ADMIN
            - SYS_ADMIN
            - SYS_RESOURCE
```

maybe we should also consider check whether they are using 5.8+ kernel then use BPF priv ...

File: Makefile [hunk 1]
```diff
@@ -90,14 +91,22 @@ cross-build: cross-build-linux-amd64 cross-build-linux-arm64
 _build_containerized: genbpfassets tidy-vendor format
 	@if [ -z '$(CTR_CMD)' ] ; then echo '!! ERROR: containerized builds require podman||docker CLI, none found $$PATH' >&2 && exit 1; fi
 	echo BIN_TIMESTAMP==$(BIN_TIMESTAMP)
+
+	rm -f $(SRC_ROOT)/build/Dockerfile.tmp
+	cp $(SRC_ROOT)/build/Dockerfile $(SRC_ROOT)/build/Dockerfile.tmp
+
+	@if [ '$(ROOTLESS)' = 'false' ] ; then echo "user root" >> $(SRC_ROOT)/build/Dockerfile.tmp; fi
```
Comment (rootfs) [inline Makefile:98]: we don't need to add `user root`. If no user is defined, the default is then root.

Comment (jichenjc) [reply Makefile:98]: ok, updated

**Comment by rootfs** [comment]
@jichenjc thanks, can you also update the doc [here](https://github.com/sustainable-computing-io/kepler-doc/blob/main/docs/installation/kepler.md#build-manifests)?

**Comment by jichenjc** [comment]
yes, will update , thanks~

**Pull request merged**